lib/tasks/compare/critical_data_changed_checker.rake

```diff
@@ -0,0 +1,233 @@
+# frozen_string_literal: true
+
+require "json-diff"
+
+Row = Struct.new(
+  :induction_record,
+  :participant_profile_id,
+  :past_lead_provider_ids,
+  :past_delivery_partner_ids,
+  :past_school_urns,
+  :past_cohort_years,
+  :past_mentor_ids,
+  :past_identity_ids,
+  :past_schedules,
+  keyword_init: true,
+) do
+  def changed_lead_provider
+    ([induction_record&.lead_provider&.id] + past_lead_provider_ids).compact.uniq.count > 1
+  end
+
+  def changed_delivery_partner
+    ([induction_record&.delivery_partner&.id] + past_delivery_partner_ids).compact.uniq.count > 1
+  end
+
+  def changed_school
+    ([induction_record&.school&.urn] + past_school_urns).compact.uniq.count > 1
+  end
+
+  def changed_cohort
+    ([induction_record&.cohort&.start_year] + past_cohort_years).compact.uniq.count > 1
+  end
+
+  def changed_mentor
+    ([induction_record&.mentor&.id] + past_mentor_ids).compact.uniq.count > 1
+  end
+
+  def changed_identity
+    ([induction_record&.preferred_identity&.external_identifier] + past_identity_ids).compact.uniq.count > 1
+  end
+
+  def changed_schedule
+    ([induction_record&.schedule&.id] + past_schedules).compact.uniq.count > 1
+  end
+
+  def changed
+    changed_lead_provider ||
+      changed_delivery_partner ||
+      changed_school ||
+      changed_cohort ||
+      changed_mentor ||
+      changed_identity ||
+      changed_schedule
```

Comment:
not important now but we can write this with `[changed_lead_provider, changed_delivery_partner, ...].any?`